mlflow/pyfunc/__init__.py

```diff
@@ -724,6 +727,85 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def _download_model_conda_env(model_uri):
+    conda_yml_file_name = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME)[ENV]
+    return _download_artifact_from_uri(os.path.join(model_uri, conda_yml_file_name))
+
+
+def get_model_dependencies(model_uri, format="pip"):  # pylint: disable=redefined-builtin
+    """
+    Given a model URL, and format ("pip" or "conda", default value "pip"),
+    return the downloaded dependency file path.
+    If "pip" format specified but model does not have "requirements.txt" file,
+    fallback to parse the pip section of the model's "conda.yaml" and ignore other
+    non-pip dependencies in the "conda.yaml" file.
+    """
+    req_file_uri = os.path.join(model_uri, _REQUIREMENTS_FILE_NAME)
+
+    if format == "pip":
+        try:
+            pip_file_path = _download_artifact_from_uri(req_file_uri)
+        except Exception:
+            # fallback to download conda.yaml file and parse the "pip" section from it.
+            _logger.info(
+                f"Download model '{_REQUIREMENTS_FILE_NAME}' file failed. Fallback to fetch "
+                "pip requirements from the model's 'conda.yaml' file and other conda "
+                "dependencies will be ignored."
+            )
+            conda_yml_path = _download_model_conda_env(model_uri)
+            pip_deps = None
+            conda_deps = []
+            try:
+                with open(conda_yml_path, "r") as yf:
+                    conda_yml = yaml.safe_load(yf)
+
+                for dep in conda_yml["dependencies"]:
+                    if (
+                        isinstance(dep, str)
+                        and not dep.startswith("python=")
+                        and not dep.startswith("pip=")
```

Comment:
```suggestion
                        and not dep.startswith("pip<=")
```

I think `<=` shoulbe be used since we don't use `=`:

https://github.com/mlflow/mlflow/blob/2e175e7fd65d86d12d204b648aa5ff73063a7dc9/mlflow/utils/environment.py#L57